### PR TITLE
Happo.io integration

### PR DESCRIPTION
**Pull request by brentertz**
### Description

Happo.io integration.  Happo is a visual regression testing service. It takes screenshots and compares them with previous versions of a component.

### Motivation and context

connects #462 

Help to prevent visual regressions.

### Screenshots, videos, or demo, if appropriate

See links to happo reports in the comments below.

### How to test

1. Get the HAPPO environment keys and set them up locally
1. Run the happo commands as described in `testing.md`
1. Make a visual change to a component, then rerun the happo command
1. Review the happo report

or

1. Create a new branch based on this branch
1. Make a visual change to a component
1. Open a test PR with this branch as the base
1. Wait for the build to complete and for the happo review to be posted on the PR
1. View the generated happo report

### Types of changes

- Other (provide details below)

Adds new visual regression testing capabilities to build system.

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support) - **[n/a]**
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered - **[n/a]**
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered - **[n/a]**
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change - **[n/a]**

### CODE REVIEW

File: utils/happo.js [hunk 1]
```diff
@@ -35,7 +48,7 @@ const allExamples = requireContext
       // eslint-disable-next-line react/display-name
       variants[id] = () => (
         <ThemeProvider>
-          <LiveProvider code={source} scope={scope} mountStylesheet>
+          <LiveProvider code={source} scope={scope} mountStylesheet={false}>
```
Comment (trotzig) [inline utils/happo.js]: Ah, didn't know we could turn that off and still have it apply styles.

Comment (brentertz) [reply utils/happo.js]: Yep, that stylesheet is just for the default prism syntax highlighting in the code examples.  We implement our own custom syntax highlighting separately.

**Comment by brentertz** [comment]
@trotzig Thanks for reviewing.  I still have a bit of additional testing to do.  Hopefully will be done today.

**Comment by brentertz** [comment]
@trotzig I believe that this PR is good to go.  Additionally, we are open to a followup meeting whenever you are.

**Comment by trotzig** [comment]
Great! I'll send you an email with a suggestion for a sync-up meeting.

File: docs/testing.md [hunk 2]
```diff
@@ -57,3 +57,23 @@ npm run jest -- --watch
 1. Open [chrome://inspect](chrome://inspect) in Chrome.
 1. Under "Remote target", click on "inspect". A new debugger window will open up.
 1. Click play once to get past the initial breakpoint.  The debugger should then stop at your breakpoint.
+
+## Visual diff tests
+
+Visual diff testing is performed using [Happo.io](https://happo.io/).
+
+Visual diff tests are run automatically on local pull-requests.  They are not run on pull-requests from repo forks as they do not have access to the respective environment variables.
```
Comment (brentertz) [inline docs/testing.md]: Is there a better word for "local"?

File: docs/testing.md [hunk 3]
```diff
@@ -57,3 +57,23 @@ npm run jest -- --watch
...
+Given that you have valid `HAPPO_KEY` and `HAPPO_SECRET` environment variables, the following commands can be used to execute visual diff tests.
+
+#### Run all visual diff tests and generate a report
+```sh
+npm run happo run
+```
+
+#### Watch for file changes and re-generate report
+```sh
+npm run happo dev
```
Comment (brentertz) [inline docs/testing.md]: Consider hiding the dev/run aspect behind an npm script

File: .happo.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  customizeWebpackConfig: (config) => {
+    config.module = {
+      rules: [
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          loader: babelLoader
+        },
+        {
+          test: /\.md$/,
```
Comment (brentertz) [inline .happo.js]: Is this strictly necessary?

**Pull request merged**